lib/segment/src/id_tracker/simple_id_tracker.rs

```diff
@@ -150,25 +151,26 @@ impl SimpleIdTracker {
         let versions_db_wrapper = DatabaseColumnScheduledUpdateWrapper::new(
             DatabaseColumnWrapper::new(store, DB_VERSIONS_CF),
         );
-        for (key, val) in versions_db_wrapper.lock_db().iter()? {
-            let external_id = Self::restore_key(&key);
-            let version: SeqNumberType = bincode::deserialize(&val).unwrap();
-            let internal_id = match external_id {
-                PointIdType::NumId(idx) => external_to_internal_num.get(&idx).copied(),
-                PointIdType::Uuid(uuid) => external_to_internal_uuid.get(&uuid).copied(),
-            };
-            if let Some(internal_id) = internal_id {
-                if internal_id as usize >= internal_to_version.len() {
-                    internal_to_version.resize(internal_id as usize + 1, 0);
-                }
-                internal_to_version[internal_id as usize] = version;
-            } else {
-                log::debug!(
-                    "Found version: {} without internal id, external id: {}",
-                    version,
-                    external_id
-                );
+        for (internal, external) in internal_to_external.iter().enumerate() {
```

Comment:
If we extend the version slice, we drop mappings that don't have any version. When loading the simple ID tracker we mark mappings missing a version as deleted.

Why is the action we take in both cases different? Can't we drop the mappings in case of loading the simple ID tracker as well?

Reply:
Yeah, it is a good point. I did not drop anything here because it affects the persistent state, while the method is only supposed to load. 

Do you think we should perform repair as well on load?

Reply:
I decided to drop the in-memory mapping without removing it from rocksdb